src/GOATalgorithm.jl

```diff
@@ -0,0 +1,152 @@
+
+"""
+    _solve_transportation_problem(data; optimizer)
+
+Given a data matrix and a JuMP optimizer, return then doubly stochastic solution of the transportation problem.
+"""
+function _solve_transportation_problem(data::AbstractMatrix{U}; optimizer) where {U<:Real}
+    M,N=size(data)
+    model = Model(optimizer)
+    set_silent(model)
+    @variable(model, x[m in 1:M, n in 1:N] >= zero(U),integer=false)
+    @objective(
+        model,
+        Min,
+        sum(data[m, n] * x[m, n] for n in 1:N, m in 1:M),
+    )
+    @constraint(model, [m in 1:M], sum(x[m, :]) == one(U))
+    @constraint(model, [n in 1:N], sum(x[:, n]) == one(U))
+    optimize!(model)
+    Q=zeros(M,N)
+    for n in 1:N
+        for m in 1:M
+            Q[m,n]=value(x[m,n])
+        end
+    end
+    return Q
+end
+
+#solve assignment problem
+"""
+    _solve_assignment_problem(data; optimizer)
+
+Given a data matrix and a JuMP optimizer, return the integer solution of the assignment problem.
+"""
+function _solve_assignment_problem(data::AbstractMatrix{U}; optimizer) where {U<:Real}
+    M,N=size(data)
+    model = Model(optimizer)
+    set_silent(model)
+    @variable(model, x[m in 1:M, n in 1:N] >= zero(U),integer=true)
+    @objective(
+        model,
+        Max,
+        sum(data[m, n] * x[m,n] for n in 1:N, m in 1:M),
+    )
+    @constraint(model, [m in 1:M], sum(x[m, :]) == one(U))
+    @constraint(model, [n in 1:N], sum(x[:, n]) == one(U))
+    optimize!(model)
+    P=zeros(M,N)
+    for n in 1:N
+        for m in 1:M
+            P[m,n]=value(x[m,n])
+        end
+    end
+    return P
+end
+
+"""
+    _step_size(A,B,P_i,Q_i)
+
+Given A,B,P_i and the direction matrix Q_i, return the step size of the gradient descent method.
+"""
+function _step_size(A::AbstractMatrix{U},B::AbstractMatrix{V},P_i::AbstractMatrix{T},Q_i::AbstractMatrix{W}) where {U<:Real,V<:Real,T<:Real,W<:Real}
+    R=P_i-Q_i
+    a= -tr(A*R*B'*R')
+    b= -tr(A*Q_i*B'*R'+A*R*B'*Q_i')
+    if a>0.
+        α=min(max(-0.5*b/a,0.),1.)
+    else
+        if a+b>0.
+            α=0.
+        else
+            α=1.
+        end
+    end
+    return α
+end
+
+"""
+    _update_P(P_i,Q_i,α)
+
+Given the previous matrix, the direction and the step size, return the updated matrix.
+"""
+function _update_P(P_i::AbstractMatrix{U},Q_i::AbstractMatrix{V},α::Float64) where {U<:Real,V<:Real}
+    P_new=α*P_i + (1-α)*Q_i
+    return P_new
+end
+
+"""
+    _check_convergence(gradient,P_i,P_new;tol)
+
+Given the gradient and two consecutive step matrices, compute if the method converged.
+"""
+function _check_convergence(gradient::AbstractMatrix{U},P_i::AbstractMatrix{V},P_new::AbstractMatrix{T};tol::Float64) where {U<:Real,V<:Real,T<:Real}
+    if norm(gradient)*(norm(P_i-P_new))<tol
+        return true
+    else
+        return false
+    end
+end
+
+"""
+    goat(A,B;optimizer,max_iter=30,tol=0.1,init+=_flat_doubly_stochastic(size(A)[1]))
```

Comment:
@aurorarossi I think it is better to call it FAQ in the end. 
In either case, it's better to specify below that 
- it is using optimal transport, unlike the original 2015 paper,
- it is not using the "lighting" version of optimal transport introduced in the GOAT paper.